### DISCUSSION TITLE
Merging OpenVDB exporter

### DISCUSSION THREAD

**Discussion by Wesxdz** [InteractiveComputerGraphics/SPlisHSPlasH#228] [Q&A]
I'm implementing an exporter for https://www.openvdb.org/
https://github.com/Wesxdz/SPlisHSPlasH/tree/openvdb

There are a few decisions related to the CMake build, particularly whether OpenVBD can be included as a submodule, included directly, or expected to be found on the system. OpenVDB 8.2 uses C++ 14 by default, which would require updating the Common build standard to support this syntax in the exporter code.

I am seeking feedback on these questions, as well as if there are any desired versions to use here. 
Happy 2023 all, hoping the new year is filled with water sims.

**Comment by janbender** [comment]
If you use OpenVDB as own library, it can be built using C++ 14 while the SPlisHSPlasH library still uses C++11. There is no need to change the version. 

I'm not sure if it is a good idea to integrate OpenVDB as submodule. The last time I compiled on Windows this was not at all straightforward. So if you integrate it as submodule, you probably give up that the library compiles on different systems.

**Comment by chunleili** [comment]
> I'm implementing an exporter for https://www.openvdb.org/ https://github.com/Wesxdz/SPlisHSPlasH/tree/openvdb
> 
> There are a few decisions related to the CMake build, particularly whether OpenVBD can be included as a submodule, included directly, or expected to be found on the system. OpenVDB 8.2 uses C++ 14 by default, which would require updating the Common build standard to support this syntax in the exporter code.
> 
> I am seeking feedback on these questions, as well as if there are any desired versions to use here. Happy 2023 all, hoping the new year is filled with water sims.

Hello,
   I read your code and found that it exports to vtk rather than ".vdb". But the title of the file is vdb. Could you explain that?

**Reply by janbender** [reply]
I don't get your point. We have currently a VTK exporter, not a vdb exporter. What do you mean by title of the file?

**Reply by Wesxdz** [reply]
Just referencing vtk for implementation. It's still a work in progress, I need to test multiple grids. I just pushed the initial density export though.

**Comment by chunleili** [comment]
> I'm implementing an exporter for https://www.openvdb.org/ https://github.com/Wesxdz/SPlisHSPlasH/tree/openvdb
> 
> There are a few decisions related to the CMake build, particularly whether OpenVBD can be included as a submodule, included directly, or expected to be found on the system. OpenVDB 8.2 uses C++ 14 by default, which would require updating the Common build standard to support this syntax in the exporter code.
> 
> I am seeking feedback on these questions, as well as if there are any desired versions to use here. Happy 2023 all, hoping the new year is filled with water sims.

And I suggest that do not put all the openvdb 8.2 lib source code into the extern directory. Because this will enlarge the repo extensively with uncessary codes. Instead, using a pakage manager(for example, vcpkg) would be better.
